### PR TITLE
Expose `calcMinimumDeposit` from cardano-api

### PR DESCRIPTION
**Pull request by Jimbo4350**
Expose `calcMinimumDeposit` to calculate the creation cost of a UTxO entry given a `Value`

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Block.hs [hunk 1]
```diff
@@ -251,6 +254,13 @@ data ChainPoint = ChainPointAtGenesis
                 | ChainPoint !SlotNo !(Hash BlockHeader)
   deriving (Eq, Show)
 
+instance ToJSON ChainPoint where
+  toJSON ChainPointAtGenesis = Aeson.String "Tip is currently at genesis block"
+  toJSON (ChainPoint slot headerHash) =
+    object [ "Slot Number" .= slot
+           , "Slot Header Hash" .= Text.decodeUtf8 (serialiseToRawBytesHex headerHash)
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/Block.hs:261]: It's unusual for keys to have spaces in them.  Is this for a specific reason?

File: cardano-cli/src/Cardano/CLI/Byron/Tx.hs [hunk 2]
```diff
@@ -181,12 +186,18 @@ nodeSubmitTx network gentx = do
...
+            localConsensusModeParams = CardanoModeParams (EpochSlots 21600)
           }
-    _res <- liftIO $ submitTxToNodeLocal connctInfo (DegenGenTx gentx)
-    --TODO: print failures
+    res <- liftIO $ submitTxToNodeLocal connctInfo (TxInByronSpecial gentx ByronEraInCardanoMode)
+    case res of
+      Net.Tx.SubmitSuccess -> liftIO $ putTextLn "Transaction successfully submitted."
+      Net.Tx.SubmitFail reason ->
+        case reason of
+          TxValidationErrorInMode err _eraInMode -> left . ByronTxSubmitError . Text.pack $ show err
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/Byron/Tx.hs:198]: What's in the second field of the error and reason for ignoring?

File: cardano-api/src/Cardano/Api/Block.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Block.hs:261]: Nop just a bad habit, I'll simplify them.

**Review by newhoggy — APPROVED** [review]

**Pull request merged**